### PR TITLE
404 on newtask for unpublished project

### PR DESCRIPTION
**Pull request by dchhabda**

### CODE REVIEW

File: pybossa/api/__init__.py [hunk 1]
```diff
@@ -186,8 +186,7 @@ def new_task(project_id):
 def _retrieve_new_task(project_id):
 
     project = project_repo.get(project_id)
-
-    if project is None:
+    if project is None or not project.published:
```
Comment (fiorda) [inline pybossa/api/__init__.py]: should this exclude admins and owners?

Comment (dchhabda) [reply pybossa/api/__init__.py]: i think no, as admin/owner would be able to submit via "test it" project option

Comment (fiorda) [reply pybossa/api/__init__.py]: doesn't "test it" go through the `newtask` route too?

Comment (dchhabda) [reply pybossa/api/__init__.py]: ah right. will fix. thanks

File: test/test_view/test_project_publish.py [hunk 2]
```diff
@@ -187,3 +187,53 @@ def test_unpublish_project(self):
...
+
+        # for unpublished project, obtaining newtask would succeed for admin
+        res = self.app.get('api/project/{}/newtask?api_key={}'
+                           .format(project.id, admin.api_key))
+        assert res.status_code == 200, 'newtask to unpublished project should succeed for admin'
+
+        # for unpublished project, obtaining newtask would succeed for subadmin
+        res = self.app.get('api/project/{}/newtask?api_key={}'
+                           .format(project.id, project.owner.api_key))
+        assert res.status_code == 200, 'newtask to unpublished project should succeed for admin'
```
Comment (fiorda) [inline test/test_view/test_project_publish.py]: to be very nitpicking it should be project owner instead of admin, similar  below

**Pull request merged**